### PR TITLE
Add Positional Search (near) with :missing modifier #510

### PR DESCRIPTION
**Pull request by prb112**
- Add support for multiple instance of the _near parameter
- Improve AND coverage tests
- Add Bounding type for missing and corresponding support for it
- Enable Search's support of :missing
- Improve Test coverage for fhir-search-tests

Signed-off-by: Paul Bastide <pbastide@us.ibm.com>

### CODE REVIEW

File: fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchDateTest.java [hunk 1]
```diff
@@ -637,7 +637,8 @@ public void testSearchDate_dateTime_Day_noTZ() throws Exception {
         assertSearchDoesntReturnSavedResource("dateTime", "ge2020-01-01");
         assertSearchDoesntReturnSavedResource("dateTime", "sa2020-01-01");
         assertSearchReturnsSavedResource(     "dateTime", "eb2020-01-01");
-        assertSearchReturnsSavedResource(     "dateTime", "ap2020-01-01");
+        // This test is very dynamic and is not reliable. 
+        // assertSearchReturnsSavedResource(     "dateTime", "ap2020-01-01");
```
Comment (lmsurpre) [inline fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchDateTest.java]: what *is* the expected behavior for `ap` on a date?

File: fhir-search/src/main/java/com/ibm/fhir/search/location/bounding/BoundingMissing.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+package com.ibm.fhir.search.location.bounding;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * This class is used to indicate a Missing search.
+ */
+public class BoundingMissing extends Bounding {
```
Comment (lmsurpre) [inline fhir-search/src/main/java/com/ibm/fhir/search/location/bounding/BoundingMissing.java]: Is this class needed?  Why isn't near:missing handled similarly to the other search parameters?

File: fhir-search/src/main/java/com/ibm/fhir/search/location/NearLocationHandler.java [hunk 3]
```diff
@@ -166,53 +168,81 @@ public BoundingBox createBoundingBox(double latitude, double longitude, double d
         List<Bounding> boundingAreas = new ArrayList<>();
         // We are only interested in the near and near-distance parameters.
```
Comment (lmsurpre) [inline fhir-search/src/main/java/com/ibm/fhir/search/location/NearLocationHandler.java]: my understanding is the `near-distance` went away and is now included as part of the `near` value...maybe update the comment to match?

File: fhir-search/src/main/java/com/ibm/fhir/search/location/bounding/BoundingMissing.java (hunk 2 above)
Comment (prb112) [reply fhir-search/src/main/java/com/ibm/fhir/search/location/bounding/BoundingMissing.java]: it is.  this is used to signal the fact that it doesn't need the 4 parameters or more related to a bounding area.

File: fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchDateTest.java (hunk 1 above)
Comment (prb112) [reply fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchDateTest.java]: it takes 10% of parameter date to NOW. 
 (or for a date, 10% of the gap between now and the date), but systems may choose other values where appropriate

It should pass now. I kicked off another build to show it's ok.

File: fhir-search/src/main/java/com/ibm/fhir/search/location/NearLocationHandler.java (hunk 3 above)
Comment (prb112) [reply fhir-search/src/main/java/com/ibm/fhir/search/location/NearLocationHandler.java]: Sure.  no worries, that was a comment I missed removing.

**Pull request merged**